lang/java/reef-io/src/main/java/org/apache/reef/io/network/impl/NetworkConnectionFactory.java

```diff
@@ -73,16 +76,27 @@
     return connection;
   }
 
-  <T> Link<NetworkConnectionServiceMessage<T>> openLink(final Identifier remoteId) throws NetworkException {
+  Link<NetworkConnectionServiceMessage<T>> openLink(final Identifier remoteId) throws NetworkException {
     return networkService.openLink(remoteId);
   }
 
-  String getConnectionFactoryId() {
-    return this.connFactoryId;
+  @Override
+  public Identifier getConnectionFactoryId() {
+    return connectionFactoryId;
+  }
+
+  @Override
+  public Identifier getEndPointId() {
+    return endPointId;
   }
 
   Identifier getSrcId() {
-    return this.networkService.getNetworkConnectionServiceId();
+    // TODO : Change to always return the endPointId when the deprecated getNetworkConnectionServiceId is removed.
```

Comment:
Ah, right. Thanks for pointing that out.

Would it make sense to replace the `TODO` to a `@deprecated in 0.13` comment? (While keeping the description.)